qt/scientific_interfaces/Indirect/IndirectSymmetriseModel.cpp

```diff
@@ -0,0 +1,67 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "IndirectSymmetriseModel.h"
+#include "IndirectDataValidationHelper.h"
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidQtWidgets/Common/AlgorithmRuntimeProps.h"
+#include "MantidQtWidgets/Common/UserInputValidator.h"
+
+#include <QDoubleValidator>
+#include <QFileInfo>
+
+using namespace IndirectDataValidationHelper;
+using namespace Mantid::API;
+
+namespace MantidQt::CustomInterfaces {
+
+//----------------------------------------------------------------------------------------------
+/** Constructor
+ */
+IndirectSymmetriseModel::IndirectSymmetriseModel() {}
+
+void IndirectSymmetriseModel::setupPreviewAlgorithm(MantidQt::API::BatchAlgorithmRunner *batchAlgoRunner,
+                                                    std::vector<long> spectraRange) {
+  // Run the algorithm on the preview spectrum only
+  IAlgorithm_sptr symmetriseAlg = AlgorithmManager::Instance().create("Symmetrise");
+  symmetriseAlg->initialize();
+  symmetriseAlg->setProperty("InputWorkspace", m_inputWorkspace);
+  symmetriseAlg->setProperty("XMin", m_eMin);
+  symmetriseAlg->setProperty("XMax", m_eMax);
+  symmetriseAlg->setProperty("SpectraRange", spectraRange);
+  symmetriseAlg->setProperty("OutputWorkspace", "__Symmetrise_temp");
```

Comment:
does this mean it wont work with alg history?

Reply:
This workspace only exists to be used in the preview window. it should not need alg history

Reply:
could we have a comment?